packages/terra-date-picker/package.json

```diff
@@ -30,7 +30,7 @@
     "moment": "^2.21.0",
     "popper.js": "^1.14.3",
```

Comment:
We'll need to work through this before we merge this PR down. Let's chat in our meeting tomorrow to figure out a plan to address these issues.

Reply:
> react-datepicker now uses react-popper. So I think we can drop popper.js.

Agreed, lets test this out.

> We need to bumper our minimum version of moment to ^2.22.0 to match react-datepicker's minimum peer dependency version

As @benbcai mentioned, react-datepicker's peer dependency range for moment is currently set to `^2.20.0`. We should be good with continuing to use `^2.21.0` as our peer dependency range for moment.

> They are using react-onclickoutside version ^6.7.1. terra-hookshot uses this component but is on version ^5.11.1. Would we want to upgrade hookshot first?

Yes, we should update hookshot first. We will likely need to polyfill classList I believe when we update `react-onclickoutside`.

> A few new props were added we may or may not want to leverage. I don't know that we would want to expose any prop relating to showing time since we have our own time-input. 

Correct, we don't want to expose any of the time related props that were added to react-datepicker.

These seem ok to enable:
`onYearChange` prop
`preventOpenOnFocus` prop
option to allow showing month navigation as disabled

Let's start on the react-onclickoutside dependency update. I believe this package is used across a few of our components. We'll want to start that work first.